modules/apps/frontend-js/frontend-js-translation-web/src/main/resources/META-INF/resources/TranslationSelector.tsx

```diff
@@ -0,0 +1,208 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import ClayButton from '@clayui/button';
+import ClayDropDown from '@clayui/drop-down';
+import ClayForm from '@clayui/form';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import ClayLayout from '@clayui/layout';
+import React from 'react';
+
+interface IItem {
+	label: string;
+	symbol: string;
+}
+
+interface ITranslations {
+	[key: string]: string;
+}
+
+export type ButtonSize = null | 'regular' | 'small';
+
+export type Variant = null | 'language' | 'translation';
+
+export interface IProps extends React.HtmlHTMLAttributes<HTMLDivElement> {
+
+	/**
+	 * Labels for the aria attributes
+	 */
+	ariaLabels?: {
+		default: string;
+		translated: string;
+		untranslated: string;
+		manageTranslations: string;
+	};
+
+	/**
+	 * The default locale
+	 */
+	defaultLocale: IItem;
+
+	/**
+	 * List of locales to allow localization for
+	 */
+	locales: Array<IItem>;
+
+	/**
+	 * Callback that gets called the "Manage Translations" drop down item is clicked
+	 */
+	onManageTranslations?: () => void;
```

Comment:
@matuzalemsteles + @diegonvs I'm probably going to remove this because we are always going to do the same thing: open the translation modal (which I'm working on right now) cc @dsanz

Reply:
The presence of that behavior (i.e. providing a "manage translation" button and opening the modal) depends on the TranslationSelector being in **admin mode** or not ([explanation](https://liferay.atlassian.net/wiki/spaces/ENGFRONTENDINFRA/pages/1593639193/LPS-133329+New+Translation+Manager#Translation-selector.-One-component%2C-multiple-roles.)).

As the translation admin is a separate component which is meant to be used on its own (and not only from the translation selector), I'd say that what translation selector could do is to compose the translation admin when invoked in admin mode, and perhaps register some listener on it so that when the user confirms the available languages, all instances of translation selector (both react and jsp-based) can update the list of available locales.

However that's an impl detail which basically means I'm not sure what I'm talking about
![image](https://user-images.githubusercontent.com/861014/123104857-98fded00-d437-11eb-9325-6ce43b55c6c7.png)

Reply:
Oh makes sense, well when I saw this I figured it would be done outside the TranslationSelector, but yes, it makes sense to remove it and add the internal implementation to the component since we have a variant for that.

Reply:
> As the translation admin is a separate component which is meant to be used on its own (and not only from the translation selector)

Thanks @dsanz that already clears some doubts: we will need the "modal" to be a separate component so that it can be used on its own, but I still think that this `onManageTranslations` callback is not needed, @matuzalemsteles let me know if that also makes sense to you when you get a chance.